core/src/components/select/select.tsx

```diff
@@ -649,6 +656,30 @@ export class Select implements ComponentInterface {
     return Array.from(this.el.querySelectorAll('ion-select-option'));
   }
 
+  /**
+   * Returns any plaintext associated with
+   * the label (either prop or slot).
+   * Note: This will not return any custom
+   * HTML. Use the `hasLabel` getter if you
+   * want to know if any slotted label content
+   * was passed.
+   */
+  private get labelText() {
+    const { el, label } = this;
+
+    if (label !== undefined) {
+      return label;
+    }
+
+    const labelSlot = el.querySelector('[slot="label"]');
+
+    if (labelSlot !== null) {
+      return labelSlot.textContent;
```

Comment:
Passing this as-is to the alert header gives odd results when using a more complicated label structure. For example, opening this:
```html
<ion-select placeholder="Select">
  <div slot="label">
    <div>Select a Fruit</div>
    <ion-note>This will be remembered.</ion-note>
  </div>
  <ion-select-option value="apple">Apple</ion-select-option>
</ion-select>
```
Leads to this:

![odd header](https://i.gyazo.com/1cb3c224d81e65e261527725f8b8a068.png)

While I can't think of a good way to have Ionic solve this automatically, I noticed you can overwrite it manually using `interfaceOptions`, and that still works:

```html
<ion-select placeholder="Select" id="weird-label">
  <div slot="label">
    <div>Select a Fruit</div>
    <ion-note>This will be remembered.</ion-note>
  </div>
  <ion-select-option value="apple">Apple</ion-select-option>
</ion-select>

<script>
  const select = document.querySelector('#weird-label');
  select.interfaceOptions = {
    header: 'Select a Fruit'
  };
</script>
```

Maybe we'll just want to document this pattern and leave it as-is?